modules/imgproc/src/featureselect.cpp

```diff
@@ -305,15 +299,36 @@ void cv::goodFeaturesToTrack( InputArray _image, OutputArray _corners,
         for( int x = 1; x < imgsize.width - 1; x++ )
         {
             float val = eig_data[x];
-            if( val != 0 && val == tmp_data[x] && (!mask_data || mask_data[x]) )
+            if( val != 0 && (fabs(val - tmp_data[x])<CV_GFTTEPS) && (!mask_data || mask_data[x]) )
```

Comment:
Float equality operation is fine here but it is not correct in general. I propose  to use this approach.


From: Alexander Alekhin [mailto:notifications@github.com]
Sent: Monday, August 10, 2015 4:23 PM
To: Itseez/opencv
Cc: Usentsov, Alexander
Subject: Re: [opencv] added test goodfeaturetotrack.cpp (#3540)


In modules/imgproc/src/featureselect.cpp<https://github.com/Itseez/opencv/pull/3540#discussion_r36629465>:

> @@ -305,15 +299,36 @@ void cv::goodFeaturesToTrack( InputArray _image, OutputArray _corners,

>          for( int x = 1; x < imgsize.width - 1; x++ )

>          {

>              float val = eig_data[x];

> -            if( val != 0 && val == tmp_data[x] && (!mask_data || mask_data[x]) )

> +            if( val != 0 && (fabs(val - tmp_data[x])<CV_GFTTEPS) && (!mask_data || mask_data[x]) )

Please revert this change. Float equality operation is fine here.

—
Reply to this email directly or view it on GitHub<https://github.com/Itseez/opencv/pull/3540/files#r36629465>.

--------------------------------------------------------------------
Closed Joint Stock Company Intel A/O
Registered legal address: Krylatsky Hills Business Park, 
17 Krylatskaya Str., Bldg 4, Moscow 121614, 
Russian Federation

This e-mail and any attachments may contain confidential material for
the sole use of the intended recipient(s). Any review or distribution
by others is strictly prohibited. If you are not the intended
recipient, please contact the sender and delete all copies.